src/server/Document/Renderers/ComponentTracking.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import processClientDeviceAndSendStaticBeacon from '#src/server/utilities/staticATITracking';
+import clickTracking from '#src/server/utilities/staticATITracking/clickTracking';
+import isOperaProxy from '#app/lib/utilities/isOperaProxy';
+import viewTracking from '#src/server/utilities/staticATITracking/viewTracking';
+
+type Props = {
+  enableStaticClickTrackingOnOperaMiniOnly: boolean;
+  trackComponentViews: boolean;
+};
+
+const trackingScripts = ({
+  enableStaticClickTrackingOnOperaMiniOnly,
+  trackComponentViews,
+}: Props) => {
+  return `
+    window.addEventListener('load', function () {
+      let isComponentTrackingEnabled = true;
+
+      if (${enableStaticClickTrackingOnOperaMiniOnly}) {
+        isComponentTrackingEnabled = ${isOperaProxy.toString()}();
+      }      
+
+      if (isComponentTrackingEnabled) {
+        (${processClientDeviceAndSendStaticBeacon.toString()})();
+
+        if (${trackComponentViews}) {
+          (${viewTracking.toString()})();
+        }
+
+        (${clickTracking.toString()})();
+      }
+    });
+  `;
+};
+
+/**
+ * Component for injecting component tracking scripts into the page.
+ *
+ * @param {Object} props - Props for the component.
+ * @param {boolean} props.enableStaticClickTrackingOnOperaMiniOnly - Enables static click tracking only on Opera Mini browsers.
+ *   If true, tracking will only occur on Opera Mini; otherwise, it will occur on all browsers.
+ * @param {boolean} props.trackComponentViews - Enables tracking of component views.
```

Comment:
Let's remove references to `props`
```suggestion
 * @param {boolean} enableStaticClickTrackingOnOperaMiniOnly - Enables static click tracking only on Opera Mini browsers.
 *   If true, tracking will only occur on Opera Mini; otherwise, it will occur on all browsers.
 * @param {boolean} trackComponentViews - Enables tracking of component views.
```